### DISCUSSION TITLE
Specifics of getting headphone sound on audiokit 2957

### DISCUSSION THREAD

**Discussion by olup** [pschatzmann/arduino-audio-tools#160] [Q&A] [answered]
Hi,
I just tried the https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-audiokit/player-url_icy-audiokit example - I have an AITHINKER audiokit 2957 board (config 6 is the only one that works).

However, I do not have any sounds coming on the earphones, only a very faint cyclic clicking sound, when really trying to hear.

I wondered if there was a special configuration that i's missed on the doc to get sound from the earphone or if I am missing something else here.

Logs from serial show that the board is online and is getting the mp3 ok. So I guess it's just a matter of outputting the sound.

**Comment by pschatzmann** [comment]
I suggest to start with a simple sine generator sketch. When you start set the log level to Info to check that the copy is working...
h

**Reply by olup** [reply]
Indeed I did this, minus the log info thing. testing now with https://github.com/pschatzmann/arduino-audiokit/blob/main/examples/headphone/output.ino - silent as well.

**Reply by olup** [reply]
https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/streams-synthbasic1-audiokit/streams-synthbasic1-audiokit.ino This sketch is silent as well. Logs seems fine, some copy happens. This is the log on clicking a button:

```
[I] SoundGenerator.h : 206 - setFrequency: 165
[I] SoundGenerator.h : 207 - active: false
[I] SoundGenerator.h : 165 - bool audio_tools::SineWaveGenerator<T>::begin() [with T = short int]
[I] SoundGenerator.h : 38 - bits_per_sample: 16
[I] SoundGenerator.h : 39 - channels: 2
[I] SoundGenerator.h : 40 - sample_rate: 44100
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[E] SoundGenerator.h : 105 - SoundGenerator::readBytes -> inactive
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 0 -> 0 bytes - in 0 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 0 -> 0 bytes - in 0 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 0 -> 0 bytes - in 0 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 0 -> 0 bytes - in 0 hops
```

However, weirdly enough, the red led in front of the jack is lighting up when I click the button key3. Others don't do that. Is there a pin configuration that is wrong on my end ? (EDIT: Seems wired this way, will light up even when sketch is on error state)

**Comment by pschatzmann** [comment]
Did you check with the sound both with a speaker and some headphones ?

The pin definitions can be found in https://github.com/pschatzmann/arduino-audiokit/blob/main/src/audio_board/ai_thinker_ac101/

- Maybe on your board the PA_ENABLE_GPIO is not 21 ? (board_def.h)
- Or it has different I2S pins (board_pins_config.c)

The 2957 boards are a big mess!

To test I recommend to use https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-audiokit/streams-generator-audiokit

**Reply by olup** [reply]
I check some sketch scanner online that confirmed I have the ac-101 version. I am not sure what to make of this information ? I'll check your link now.
`ES-8388 I2C device found at address 0x1A`

**Comment by pschatzmann** [comment]
On some boards the following is needed: 
        i2s_config->bck_io_num = GPIO_NUM_5; // GPIO_NUM_27

**Reply by olup** [reply]
may I ask where this should go ? In my sketch's setup block ? in the lib ?

**Comment by pschatzmann** [comment]
In the audio-kit lib src/audio_board/ai_thinker_ac101/board_pins_config.c

**Reply by olup** [reply]
no dice, alas

**Comment by olup** [comment]
Ok, I think I got it to work. Thing is, even with `#define AUDIOKIT_BOARD 6` (and a log telling me I chose this board) the `board_pins_config.c` file used is always the one from `ai_thinker_es8388_3478`. I must do something wrong. Once overridden the pins declaration in this file from the values found in the ic101 definition, it went smooth.

**Comment by pschatzmann** [comment, marked as answer]
Did you really define the board  in src/AudioKitSettings.h as descirbed in the Readme ?
To do this in the sketch will not work...

**Reply by olup** [reply]
So I learned... Sorry for wasting your time !

**Comment by pschatzmann** [comment]
I will update the Readme to warn about this...

**Reply by olup** [reply]
Thanks, and thanks again for this invaluable library
